src/server/routes/revisionsList.js

```diff
@@ -0,0 +1,145 @@
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as propHelpers from '../../client/helpers/props';
+import * as utils from '../helpers/utils';
+
+import {escapeProps, generateProps} from '../helpers/props';
+
+import AboutPage from '../../client/components/pages/about';
+import ContributePage from '../../client/components/pages/contribute';
+import DevelopPage from '../../client/components/pages/develop';
+import HelpPage from '../../client/components/pages/help';
+import Layout from '../../client/containers/layout';
+import LicensingPage from '../../client/components/pages/licensing';
+import PrivacyPage from '../../client/components/pages/privacy';
+import React from 'react';
+import ReactDOMServer from 'react-dom/server';
+import RevisionsListPage from '../../client/components/pages/revisionsListPage';
+import _ from 'lodash';
+import express from 'express';
+import target from '../templates/target';
+
+
+const router = express.Router();
+
+/* GET home page. */
+router.get('/', async (req, res, next) => {
+	const {orm} = req.app.locals;
+	const size = req.query.size ? parseInt(req.query.size, 10) : 20;
+	const from = req.query.from ? parseInt(req.query.from, 10) : 0;
+
+	function render(entities) {
+		const props = generateProps(req, res, {
+			from,
+			recent: _.take(entities, size),
+			size
+		});
+
+		/*
+		 * Renders react components server side and injects markup into target
+		 * file object spread injects the app.locals variables into React as
+		 * props
+		 */
+		const markup = ReactDOMServer.renderToString(
+			<Layout {...propHelpers.extractLayoutProps(props)}>
+				<RevisionsListPage
+					// disableSignUp={req.signUpDisabled}
+					from={props.from}
+					results={props.recent}
+					size={props.size}
+				/>
+			</Layout>
+		);
+
+		res.send(target({
+			dev: process.env.NODE_ENV === 'development',
+			markup,
+			props: escapeProps(props),
+			script: '/js/revisionsListPage.js',
+			title: 'Revisions Page'
+		}));
+	}
+
+	const entityModels = utils.getEntityModels(orm);
+
+	try {
+		const orderedEntities = await utils.getOrderedEntities(from, size, entityModels, orm);
+		return render(orderedEntities);
+	}
+	catch (err) {
+		return next(err);
+	}
+});
+
+
+router.get('/revisions', async (req, res, next) => {
+	const {orm} = req.app.locals;
+	const size = req.query.size ? parseInt(req.query.size, 10) : 20;
+	const from = req.query.from ? parseInt(req.query.from, 10) : 0;
+
+	function render(entities) {
+		const props = generateProps(req, res, {
+			from,
+			recent: _.take(entities, size),
+			size
+		});
+		res.send(props.recent);
+	}
+
+	const entityModels = utils.getEntityModels(orm);
+
+	try {
+		const orderedEntities = await utils.getOrderedEntities(from, size, entityModels, orm);
+		return render(orderedEntities);
```

Comment:
Here you can do `res.send(_.take(orderedEntities,size);` instead (no need for `return`in this case), and remove the `render` function